src/store/auth/sagas.ts

```diff
@@ -102,15 +102,61 @@ function * loginUserWorker () {
     const user = profile.user;
     const userId = user.id;
     const userName = user.name.split(" ");
-    const currentOrg = profile.current_org;
+
+    let currentOrg = {
+      id: "",
+      name: "",
+      role: {
+        id: "",
+        name: ROLES.baseUser.value,
+      },
+    };
+
+    const currentUser: string | null = localStorage.getItem("currentUser");
+
+    if (currentUser && currentUser !== userId) {
+      localStorage.removeItem("orgInStorage");
+      
+      localStorage.setItem("currentUser", userId);
+    }
+
+    const orgInStorage: string | null = localStorage.getItem("orgInStorage");
+
+    if (orgInStorage) {
+      const parsedOrg = JSON.parse(orgInStorage);
+
+      currentOrg = {
+        id: parsedOrg.id,
+        name: parsedOrg.name,
+        role: {
+          id: parsedOrg.role.id,
+          name: parsedOrg.role.name,
+        },
+      };
+    }
+    
+    if (!orgInStorage && profile.organizations.length) {
```

Comment:
Not much of a difference but since `!orgInStorage! is the opposite of the previous condition, this could be `else if (profile.organizations.length) {`, which slightly reduces a chance of someone making a mistake in the future since the code does not return in the first if.